tests/js/server/aql/aql-graph-traverser.js

```diff
@@ -2412,6 +2412,320 @@ function complexFilteringSuite() {
+      let query = `WITH ${vn} FOR v,e,p IN 1..10 OUTBOUND @start @eCol PRUNE pruneCondition = ${condition} OPTIONS {bfs: true} FILTER pruneCondition RETURN {from: e._from, to: e._to}`;
+      let bindVars = {
+        'eCol': en,
+        'start': vertex.A
+      };
+      let queryExplain = AQL_EXPLAIN(query, bindVars).plan.nodes; //check for extra LET
+      let foundExtraLet = false;
+      queryExplain.forEach((node) => { 
+        if(node.type === "CalculationNode" && node["outVariable"]["name"] === "pruneCondition") {
+          foundExtraLet = true;
+        }
+        if(node.type === "FilterNode") {
+          assertEqual(node["inVariable"]["name"], "pruneCondition");
+        }
+        if(node.type === "TraversalNode") {
+          assertEqual(node["options"]["order"], "bfs");
+        }
+      });
+      assertTrue(foundExtraLet);
+    },
+
+    testStorePruneConditionPathInVariableExplain: function () {
+      // note: using NOOPT(...) here to disable the optimization that pulls
+      // certain FILTER conditions into the traversal
+      const condition = "NOOPT(p.edges[0].right) == true";
+      let query = `WITH ${vn} FOR v,e,p IN 1..10 OUTBOUND @start @eCol PRUNE pruneCondition = ${condition} FILTER pruneCondition RETURN {from: e._from, to: e._to}`;
+      let bindVars = {
+        'eCol': en,
+        'start': vertex.A
+      };
+      let queryExplain = AQL_EXPLAIN(query, bindVars).plan.nodes; //check for extra LET
+      let foundExtraLet = false;
+      queryExplain.forEach((node) => { 
+        if(node.type === "CalculationNode" && node["outVariable"]["name"] === "pruneCondition") {
+          foundExtraLet = true;
+        }
+        if(node.type === "FilterNode") {
+          assertEqual(node["inVariable"]["name"], "pruneCondition");
+        }
+      });
+      assertTrue(foundExtraLet);
+    },
+
+    testStorePruneConditionPathInVariableExplainWithOptions: function () {
+      // note: using NOOPT(...) here to disable the optimization that pulls
+      // certain FILTER conditions into the traversal
+      const condition = "NOOPT(p.edges[0].right) == true";
+      let query = `WITH ${vn} FOR v,e,p IN 1..10 OUTBOUND @start @eCol PRUNE pruneCondition = ${condition} OPTIONS {bfs: true} FILTER pruneCondition RETURN {from: e._from, to: e._to}`;
+      let bindVars = {
+        'eCol': en,
+        'start': vertex.A
+      };
+      let queryExplain = AQL_EXPLAIN(query, bindVars).plan.nodes; //check for extra LET
+      let foundExtraLet = false;
+      queryExplain.forEach((node) => { 
+        if(node.type === "CalculationNode" && node["outVariable"]["name"] === "pruneCondition") {
+          foundExtraLet = true;
+        }
+        if(node.type === "FilterNode") {
+          assertEqual(node["inVariable"]["name"], "pruneCondition");
+        }
+        if(node.type === "TraversalNode") {
+          assertEqual(node["options"]["order"], "bfs");
+        }
+      });
+      assertTrue(foundExtraLet);
+    },
```

Comment:
This Test I think is very important, so even if you would not do any of the user suggested test I would like to see at least this one here.

Test the Scope of the PRUNE condition.

Try to use the following Query:
```
FOR v IN 1 OUTBOUND @start PRUNE pruneCondition = v.right == true
/* Collect will break the scope and needs to invalidate all variables, this collect however is a "pass through" as _key is always unique */
COLLECT tmp = v._key
/* Here v is not defined anymore*/
/* if the PRUNE condition is simply written down, this can fail because v is undefined */
RETURN pruneCondition
```

And also this query (change only after COLLECT):

```
FOR v IN 1 OUTBOUND @start PRUNE pruneCondition = v.right == true
/* Collect will break the scope and needs to invalidate all variables, this collect however is a "pass through" as _key is always unique */
COLLECT tmp = v._key
/* Here v is not defined anymore*/
/* We redefine v to something unrelated */
LET v = {right: true}
/* Now the pruneCondition has a v to read, however it is totally unrelated to the original v as used by the condition.*/
RETURN pruneCondition
```

Both of the above queries should fail in my opinion as the pruneCondition needs to be invalidated by COLLECT.

Reply:
Added these suggestions in the last commit